### PR TITLE
Add message delete route

### PR DESCRIPTION
**Pull request by abnegate**
## What does this PR do?

- Add delete message endpoint

## Test Plan

- Added test for delete message cases

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/controllers/api/messaging.php [hunk 1]
```diff
@@ -3266,3 +3266,51 @@
...
+    ->desc('Delete a message')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'message.delete')
+    ->label('audits.resource', 'message/{request.route.messageId}')
+    ->label('event', 'messages.[messageId].delete')
+    ->label('scope', 'messages.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'delete')
+    ->label('sdk.description', '/docs/references/messaging/delete-message.md')
```
Comment (stnguyen90) [inline app/controllers/api/messaging.php:3280]: Shouldn't this PR include this file?

File: app/controllers/api/messaging.php [hunk 2]
```diff
@@ -3266,3 +3266,51 @@
...
+        $message = $dbForProject->getDocument('messages', $messageId);
+
+        if ($message->isEmpty()) {
+            throw new Exception(Exception::MESSAGE_NOT_FOUND);
+        }
+
+        switch ($message->getAttribute('status')) {
+            case MessageStatus::PROCESSING:
+                throw new Exception(Exception::MESSAGE_ALREADY_SCHEDULED);
+            case MessageStatus::SCHEDULED:
```
Comment (stnguyen90) [inline app/controllers/api/messaging.php:3298]: Should we also check if the status is scheduled and `scheduledAt` is after now?

File: app/controllers/api/messaging.php (hunk 1 above)
Comment (abnegate) [reply app/controllers/api/messaging.php:3280]: Added

File: app/controllers/api/messaging.php (hunk 2 above)
Comment (abnegate) [reply app/controllers/api/messaging.php:3298]: Yep makes sense, added

**Review by stnguyen90 — APPROVED** [review]

**Pull request merged**